gen

```diff
@@ -20,12 +20,19 @@
 
 version="Generate (gen) version: 1.0"
 
+ext=$(echo ${1##*.})
+templates=templates/template.$ext
```

Comment:
I want this feature. How can we make it clearer. Just floating out at the top of the script isn't ideal.

Reply:
Hmm... Just explain how it works would suffice i think. If no template file is found, the user wouldn't notice any difference.

I think it's a nice way of integrating templates/boilerplate script when creating new files.

You could have different templates in separate (project) folders, and all you have to do is `cd` to that folder and create a new script, and the template is automatically loaded.

Thought about the best way to integrate this without loosing the portability, and this is basically what i ended up with, since if you move the functions out in templates, the main script wouldn't function without them, which isn't ideal.

Example templates: [natelandau/shell-scripting-templates](https://github.com/natelandau/shell-scripting-templates)

Perhaps make a function switch, if no template use default, and if template, use template, then the template can contain the shebang line. :+1: